pkg/dyff/reports.go

```diff
@@ -1,30 +1,93 @@
 package dyff
 
-import "github.com/gonvenience/ytbx"
+import "regexp"
+
+func (r Report) filter(f func(string) bool) (result Report) {
+	result = Report{
+		From: r.From,
+		To:   r.To,
+	}
+
+	for _, diff := range r.Diffs {
+		diffPathString := diff.Path.String()
+		if f(diffPathString) {
+			result.Diffs = append(result.Diffs, diff)
+		}
+	}
+
+	return result
+}
 
 // Filter accepts YAML paths as input and returns a new report with differences for those paths only
-func (r Report) Filter(paths ...*ytbx.Path) (result Report) {
+func (r Report) Filter(paths ...string) (result Report) {
 	if len(paths) == 0 {
 		return r
 	}
 
-	result = Report{
-		From: r.From,
-		To:   r.To,
+	return r.filter(func(s string) bool {
+		for _, path := range paths {
+			if path == s {
```

Comment:
This is causing a [regression when using Dot style paths](https://github.com/homeport/dyff/blob/7dd47ecae7ed9ed304cc7fd8f49ff3e32766ad70/internal/cmd/cmds_test.go#L511-L515), e.g. `yaml.foo.bar` instead of `/yaml/foo/bar`. I played around with your code and got it to work with the following file:
```go
package dyff

import (
	"regexp"

	"github.com/gonvenience/ytbx"
)

func (r Report) filter(hasPath func(*ytbx.Path) bool) (result Report) {
	result = Report{
		From: r.From,
		To:   r.To,
	}

	for _, diff := range r.Diffs {
		if hasPath(diff.Path) {
			result.Diffs = append(result.Diffs, diff)
		}
	}

	return result
}

// Filter accepts YAML paths as input and returns a new report with differences for those paths only
func (r Report) Filter(paths ...string) (result Report) {
	if len(paths) == 0 {
		return r
	}

	return r.filter(func(filterPath *ytbx.Path) bool {
		for _, pathString := range paths {
			path, err := ytbx.ParsePathStringUnsafe(pathString)
			if err == nil && path.String() == filterPath.String() {
				return true
			}
		}

		return false
	})
}

// Exclude accepts YAML paths as input and returns a new report with differences without those paths
func (r Report) Exclude(paths ...string) (result Report) {
	if len(paths) == 0 {
		return r
	}

	return r.filter(func(filterPath *ytbx.Path) bool {
		for _, pathString := range paths {
			path, err := ytbx.ParsePathStringUnsafe(pathString)
			if err == nil && path.String() == filterPath.String() {
				return false
			}
		}

		return true
	})
}

// FilterRegexp accepts regular expressions as input and returns a new report with differences for matching those patterns
func (r Report) FilterRegexp(pattern ...string) (result Report) {
	if len(pattern) == 0 {
		return r
	}

	regexps := make([]*regexp.Regexp, len(pattern))
	for i := range pattern {
		regexps[i] = regexp.MustCompile(pattern[i])
	}

	return r.filter(func(filterPath *ytbx.Path) bool {
		for _, regexp := range regexps {
			if regexp.MatchString(filterPath.String()) {
				return true
			}
		}
		return false
	})
}

// ExcludeRegexp accepts regular expressions as input and returns a new report with differences for not matching those patterns
func (r Report) ExcludeRegexp(pattern ...string) (result Report) {
	if len(pattern) == 0 {
		return r
	}

	regexps := make([]*regexp.Regexp, len(pattern))
	for i := range pattern {
		regexps[i] = regexp.MustCompile(pattern[i])
	}

	return r.filter(func(filterPath *ytbx.Path) bool {
		for _, regexp := range regexps {
			if regexp.MatchString(filterPath.String()) {
				return false
			}
		}
		return true
	})
}
```

Since there are deletions and new lines, I cannot use the suggestions feature in a good way.